ci/test-sriov-secondary-network-aws.sh

```diff
@@ -0,0 +1,552 @@
+        sudo apt-mark hold kubelet kubeadm kubectl
+        sudo swapoff -a
+        sudo sed -i '/ swap / s/^\(.*\)$/#\1/g' /etc/fstab
+EOF
+}
+
+# Function to initialize the Kubernetes control-plane node
+initialize_control_plane_node() {
+    local control_plane_node_ip=$1
+    echo "Initializing Kubernetes control-plane node on $control_plane_node_ip..."
+
+    ssh -o StrictHostKeyChecking=no -i "$AWS_EC2_SSH_KEY_NAME" ubuntu@"$control_plane_node_ip" << EOF
+        sudo kubeadm init --pod-network-cidr=10.244.0.0/16
+        mkdir -p \$HOME/.kube
+        sudo cp -i /etc/kubernetes/admin.conf \$HOME/.kube/config
+        sudo chown \$(id -u):\$(id -g) \$HOME/.kube/config
+EOF
+}
+
+# Function to get the join command for the worker node
+get_join_command() {
+    local control_plane_node_ip=$1
+    join_command=$(ssh -o StrictHostKeyChecking=no -i "$AWS_EC2_SSH_KEY_NAME" ubuntu@"$control_plane_node_ip" "sudo kubeadm token create --print-join-command")
+    echo "$join_command"
+}
+
+# Function to join the worker node to the cluster
+join_worker() {
+    local worker_ip=$1
+    local join_command=$2
+    echo "Joining worker node with IP $worker_ip to the Kubernetes cluster..."
+
+    ssh -o StrictHostKeyChecking=no -i "$AWS_EC2_SSH_KEY_NAME" ubuntu@$worker_ip << EOF
+        sudo $join_command
+EOF
+}
+
+# Function to verify the Kubernetes cluster status
+verify_cluster() {
+    local control_plane_node_ip=$1
+    echo "Verifying Kubernetes cluster status on control-plane node $control_plane_node_ip..."
+
+    ssh -o StrictHostKeyChecking=no -i "$AWS_EC2_SSH_KEY_NAME" ubuntu@$control_plane_node_ip << EOF
+        kubectl get nodes
+EOF
+}
+
+function setup_cluster {
+    # Launch Master and Worker EC2 instances
+    echo "Launching EC2 instance ..."
+    CONTROLPLANE_INSTANCE_ID=$(launch_ec2_instance "ControlPlane")
+    echo "ControlPlane EC2 instance launched with Instance ID: $CONTROLPLANE_INSTANCE_ID"
+    WORKER_INSTANCE_ID=$(launch_ec2_instance "Worker")
+    echo "Worker EC2 instance launched with Instance ID: $WORKER_INSTANCE_ID"
+
+    attach_network_interface "$CONTROLPLANE_INSTANCE_ID" "control-plane"
+    attach_network_interface "$WORKER_INSTANCE_ID" "worker"
+
+    echo "======== CONTROLPLANE_INSTANCE_ID: $CONTROLPLANE_INSTANCE_ID, WORKER_INSTANCE_ID: $WORKER_INSTANCE_ID ========="
+
+    # Get the public IP addresses of the instances
+    CONTROLPLANE_IP=$(get_instance_ip "$CONTROLPLANE_INSTANCE_ID")
+    echo "Get PublicIpAddress of EC2 Instance $CONTROLPLANE_INSTANCE_ID: $CONTROLPLANE_IP"
+    WORKER_IP=$(get_instance_ip "$WORKER_INSTANCE_ID")
+    echo "Get PublicIpAddress of EC2 Instance $WORKER_INSTANCE_ID: $WORKER_IP"
+
+    echo "====== CONTROLPLANE_IP: $CONTROLPLANE_IP, WORKER_IP: $WORKER_IP ======"
+
+    # Wait for EC2 instances to be fully ready
+    echo "Waiting for EC2 instances to be ready..."
+    sleep 120  # You may adjust the sleep time based on your environment
```

Comment:
That's a long sleep. Is there any aws CLI command that we can use, or can we periodically try pinging / ssh'ing into the instances?

Reply:
Updated.